docs/developer-guide/working-with-rules.md

```diff
@@ -306,13 +306,13 @@ Here, the `fix()` function is used to insert a semicolon after the node. Note th
 The `fixer` object has the following methods:
 
 * `insertTextAfter(nodeOrToken, text)` - inserts text after the given node or token
-* `insertTextAfterRange(range, text)` - inserts text after the given range
+* `insertTextAfterRange(nodeOrToken.range, text)` - inserts text after the given range
 * `insertTextBefore(nodeOrToken, text)` - inserts text before the given node or token
-* `insertTextBeforeRange(range, text)` - inserts text before the given range
+* `insertTextBeforeRange(nodeOrToken.range, text)` - inserts text before the given range
 * `remove(nodeOrToken)` - removes the given node or token
-* `removeRange(range)` - removes text in the given range
+* `removeRange(nodeOrToken.range)` - removes text in the given range
 * `replaceText(nodeOrToken, text)` - replaces the text in the given node or token
-* `replaceTextRange(range, text)` - replaces the text in the given range
+* `replaceTextRange(nodeOrToken.range, text)` - replaces the text in the given range
```

Comment:
I think it's better to describe data structure.

`*Range` methods are commonly used when the range isn't directly taken from a single node/token, but calculated. Most often from other token's ranges, like in [this example](https://github.com/eslint/eslint/blob/6f4540ea7ea39775906526506fd7abd7ea97610c/lib/rules/array-bracket-spacing.js#L97), but it also doesn't have to be related to any nodes/tokens, as in [this example](https://github.com/eslint/eslint/blob/6f4540ea7ea39775906526506fd7abd7ea97610c/lib/rules/no-multiple-empty-lines.js#L141).

If we want to replace a single node/token, we can just call `replaceText(nodeOrToken, text)`.

Reply:
Yeah, that’s a good point. The range is just a two-item array of the starting index and ending index, which exists naturally on nodes or tokens as a range property but can be manually created too.